### PR TITLE
dist/etc/conf: Add nucleo-wl55jc

### PR DESCRIPTION
**Pull request by akshaim**
This PR adds ST Nucleo-wl55jc env for HIL testing.

Reference : 
[Board Schematic](https://www.st.com/resource/en/user_manual/dm00622917-stm32wl-nucleo64-board-mb1389-stmicroelectronics.pdf)

### CODE REVIEW

File: dist/etc/conf/nucleo-wl55jc.env [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+HIL_DUT_DAC_PORT?=
+HIL_DUT_DAC_PIN?=
+
+HIL_DUT_ADC_GPIO?="PB1/CN8-1/A0"
+HIL_DUT_ADC_PORT?=1
+HIL_DUT_ADC_PIN?=1
+
+HIL_DUT_IC_GPIO?=
+HIL_DUT_IC_PORT?=
+HIL_DUT_IC_PIN?=
```
Comment (MrKevinWeiss) [inline dist/etc/conf/nucleo-wl55jc.env:59]: The IC pin is important for the timing benchmarks:

I would recommend maybe:
```suggestion
HIL_DUT_IC_GPIO?="PB8/CN9-3/D5"
HIL_DUT_IC_PORT?=1
HIL_DUT_IC_PIN?=8
```

File: dist/etc/conf/nucleo-wl55jc.env [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+HIL_DUT_PWM_PORT?=1
+HIL_DUT_PWM_PIN?=3
+
+HIL_DUT_DAC_GPIO?=
+HIL_DUT_DAC_PORT?=
+HIL_DUT_DAC_PIN?=
+
+HIL_DUT_ADC_GPIO?="PB1/CN8-1/A0"
+HIL_DUT_ADC_PORT?=1
+HIL_DUT_ADC_PIN?=1
```
Comment (MrKevinWeiss) [inline dist/etc/conf/nucleo-wl55jc.env:55]: This is for PHiLIP ADC (meaning you would feed in a DAC)

I cannot see one so maybe it is better to leave it blank:

```suggestion
HIL_DUT_ADC_GPIO?=
HIL_DUT_ADC_PORT?=
HIL_DUT_ADC_PIN?=
```

File: dist/etc/conf/nucleo-wl55jc.env [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+HIL_DUT_SDA_PORT?=0
+HIL_DUT_SDA_PIN?=11
+
+HIL_DUT_PWM_GPIO?="PB3/CN9-5/D3"
+HIL_DUT_PWM_PORT?=1
+HIL_DUT_PWM_PIN?=3
+
+HIL_DUT_DAC_GPIO?=
+HIL_DUT_DAC_PORT?=
+HIL_DUT_DAC_PIN?=
```
Comment (MrKevinWeiss) [inline dist/etc/conf/nucleo-wl55jc.env:51]: See the below comment but this is where the nucleo ADC should connect to:

```suggestion
HIL_DUT_DAC_GPIO?="PB1/CN8-1/A0"
HIL_DUT_DAC_PORT?=1
HIL_DUT_DAC_PIN?=1
```

**Review by MrKevinWeiss — COMMENTED** [review]
This should be good enough to do some HiL tests, I will try to get it wired as soon as I clear out some of the backlog.

Many thanks!

**Comment by akshaim** [comment]
@MrKevinWeiss  Thanks for the comments :)

**Comment by MrKevinWeiss** [comment]
I have it wired but cannot build anything it seems.  Once that is sorted I can run the tests.

File: dist/etc/conf/nucleo-wl55jc.env [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+HIL_DUT_GPIO0_GPIO?="PC2/CN5-1/D8"
+HIL_DUT_GPIO0_PORT?=2
+HIL_DUT_GPIO0_PIN?=2
+
+HIL_DUT_GPIO1_GPIO?="PC1/CN9-1/D7"
+HIL_DUT_GPIO1_PORT?=2
+HIL_DUT_GPIO1_PIN?=1
+
+HIL_DUT_GPIO2_GPIO?="PB5/CN9-4/D4"
```
Comment (MrKevinWeiss) [inline dist/etc/conf/nucleo-wl55jc.env:69]: Somehow this is wrong too `B5/CN9-4/D4` the location seems fine maybe it is the PB5?

File: dist/etc/conf/nucleo-wl55jc.env [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+HIL_DUT_CTS_GPIO?=
+HIL_DUT_CTS_PORT?=
+HIL_DUT_CTS_PIN?=
+
+HIL_DUT_RTS_GPIO?=
+HIL_DUT_RTS_PORT?=
+HIL_DUT_RTS_PIN?=
+
+HIL_DUT_RX_GPIO?="PA2/CN9-7/D1"
```
Comment (MrKevinWeiss) [inline dist/etc/conf/nucleo-wl55jc.env:13]: Are you sure this uart is not being used?

The fact that I get a timeout makes me suspect it is not just a wiring issue.

File: dist/etc/conf/nucleo-wl55jc.env [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+HIL_DUT_RX_PORT?=0
+HIL_DUT_RX_PIN?=2
+
+HIL_DUT_TX_GPIO?="PA3/CN9-8/D0"
```
Comment (MrKevinWeiss) [inline dist/etc/conf/nucleo-wl55jc.env:17]: Are you sure this uart is not being used?

**Review by MrKevinWeiss — CHANGES_REQUESTED** [review]
Below is the test report from the gpio test to ensure wiring, other than the 
SCL which was my fault I think there are some things to double check.

I would also like the labels of CN10 added as the connections to these are much cleaner to wire (I don't require an additional gender changing pin).
<details>

```
==============================================================================
tests_periph_gpio                                                             
==============================================================================
tests_periph_gpio.Periph Gpio Base :: Verify if board to philip wiring is c...
==============================================================================
Verify GPIO_0                                                         | PASS |
------------------------------------------------------------------------------
Verify GPIO_1                                                         | FAIL |
1 != 0
------------------------------------------------------------------------------
Verify GPIO_2                                                         | FAIL |
1 != 0
------------------------------------------------------------------------------
tests_periph_gpio.Periph Gpio Base :: Verify if board to philip wi... | FAIL |
3 tests, 1 passed, 2 failed
==============================================================================
tests_periph_gpio.Periph Gpio Others :: Verify if board to philip wiring is...
==============================================================================
Verify UART_CTS                                                       | SKIP |
Test failed but its tags matched '--SkipOnFailure' and it was marked skipped.

Original failure:
This Periph is not supported here
------------------------------------------------------------------------------
Verify UART_RTS                                                       | SKIP |
Test failed but its tags matched '--SkipOnFailure' and it was marked skipped.

Original failure:
This Periph is not supported here
------------------------------------------------------------------------------
Verify UART_RX                                                        | FAIL |
'Timeout' does not contain 'Success'
------------------------------------------------------------------------------
Verify UART_TX                                                        | FAIL |
'Timeout' does not contain 'Success'
------------------------------------------------------------------------------
Verify SPI_NSS                                                        | PASS |
------------------------------------------------------------------------------
Verify SPI_SCK                                                        | PASS |
------------------------------------------------------------------------------
Verify SPI_MISO                                                       | PASS |
------------------------------------------------------------------------------
Verify SPI_MOSI                                                       | PASS |
------------------------------------------------------------------------------
Verify I2C_SCL                                                        | FAIL |
1 != 0
------------------------------------------------------------------------------
Verify I2C_SDA                                                        | PASS |
------------------------------------------------------------------------------
Verify ADC                                                            | SKIP |
Test failed but its tags matched '--SkipOnFailure' and it was marked skipped.

Original failure:
This Periph is not supported here
------------------------------------------------------------------------------
Verify PWM                                                            | FAIL |
1 != 0
------------------------------------------------------------------------------
Verify IC                                                             | FAIL |
1 != 0
------------------------------------------------------------------------------
tests_periph_gpio.Periph Gpio Others :: Verify if board to philip ... | FAIL |
13 tests, 5 passed, 5 failed, 3 skipped
==============================================================================
tests_periph_gpio.Periph Gpio Reset :: Verify if board to philip reset wiri...
==============================================================================
Verify Reset Pin Is Connected :: Lock the dut and try run another ... | PASS |
------------------------------------------------------------------------------
tests_periph_gpio.Periph Gpio Reset :: Verify if board to philip r... | PASS |
1 test, 1 passed, 0 failed
==============================================================================
tests_periph_gpio                                                     | FAIL |
17 tests, 7 passed, 7 failed, 3 skipped

```

File: dist/etc/conf/nucleo-wl55jc.env [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+HIL_DUT_IC_GPIO?="PB8/CN9-3/D5"
+HIL_DUT_IC_PORT?=1
+HIL_DUT_IC_PIN?=8
+
+HIL_DUT_GPIO0_GPIO?="PC2/CN5-1/D8"
+HIL_DUT_GPIO0_PORT?=2
+HIL_DUT_GPIO0_PIN?=2
+
+HIL_DUT_GPIO1_GPIO?="PC1/CN9-1/D7"
```
[the remark this answers is not in the document]
Comment (akshaim) [reply dist/etc/conf/nucleo-wl55jc.env:65]: Datasheet
The datasheet says so.

File: dist/etc/conf/nucleo-wl55jc.env (hunk 4 above)
Comment (akshaim) [reply dist/etc/conf/nucleo-wl55jc.env:69]: Can you please check this image? I referred to the same. 
Datasheet

File: dist/etc/conf/nucleo-wl55jc.env (hunk 6 above)
Comment (akshaim) [reply dist/etc/conf/nucleo-wl55jc.env]: Oops. Seems like the default PIN mapping was different. Updated in fixup

File: dist/etc/conf/nucleo-wl55jc.env (hunk 5 above)
Comment (akshaim) [reply dist/etc/conf/nucleo-wl55jc.env]: > Below is the test report from the gpio test to ensure wiring, other than the
> SCL which was my fault I think there are some things to double check.
> 
> I would also like the labels of CN10 added as the connections to these are much cleaner to wire (I don't require an additional gender changing pin).

Okay. I have added references to Morpho connectors in latest commit.

File: dist/etc/conf/nucleo-wl55jc.env (hunk 7 above)
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply dist/etc/conf/nucleo-wl55jc.env]: Yup datasheet must have the pins wrong (pin 1 should start on D0 label but it seems like they did it the other way around)...

**Review by MrKevinWeiss — APPROVED** [review]
Tested on a local node and it looks good.

**Comment by akshaim** [comment]
Thanks for the review. :)

**Pull request merged**